pkg/build/gobuild.go

```diff
@@ -241,8 +239,51 @@ func getGoarm(platform v1.Platform) (string, error) {
 	return "", nil
 }
 
-func build(ctx context.Context, ip string, dir string, platform v1.Platform, config Config) (string, error) {
-	buildArgs, err := createBuildArgs(config)
+type executableBuilder interface {
+	BuildExecutable(ctx context.Context, importPath string, config Config) (exectuablePath string, err error)
+}
+
+type tinygoExecutableBuilder struct {
+	dir string
+}
+
+func (b *tinygoExecutableBuilder) BuildExecutable(ctx context.Context, ip string, config Config) (string, error) {
```

Comment:
gobuild.go is getting _huge_ (which is my fault, not yours). Could we move these wasi-specific things into a new pkg/build/wasi.go?

Reply:
Done.